dotenv-linter/src/schema.rs

```diff
@@ -0,0 +1,384 @@
+use serde::Deserialize;
+use std::{collections::HashMap, fs::File, io::BufReader, path::Path};
+
+#[derive(Deserialize, Default)]
+#[serde(default)]
+struct ReadDotEnvSchema {
+    pub version: String,
+    pub allow_other_keys: bool,
+    pub entries: Vec<SchemaEntry>,
+}
+
+pub struct DotEnvSchema {
+    pub version: String,
+    pub allow_other_keys: bool,
+    pub entries: HashMap<String, SchemaEntry>,
+}
+#[derive(Deserialize, Default)]
+#[serde(default)]
+
+pub struct SchemaEntry {
+    pub key: String,
+    pub required: bool,
+    #[serde(rename = "type")]
+    pub value_type: SchemaValueType,
+    pub regex: Option<String>,
```

Comment:
```suggestion
    pub regex: Option<regex::Regex>,
```
I think it's best to store the already parsed regex in this field.